<robot name="maverick" version="1.0">
  <joint name="left_end_effector_joint" type="fixed">
    <origin rpy="-0.00000 0.00000 -0.00000" xyz="0.12600 -0.00000 0.00000"/>
    <parent link="left_link5"/>
    <child link="left_end_effector_link"/>
  </joint>
  <joint name="left_gripper" type="prismatic">
    <limit lower="-0.01000" upper="0.01900" effort="1" velocity="4.80000"/>
    <origin rpy="-0.00000 0.00000 -0.00000" xyz="0.08170 0.02100 0.00000"/>
    <parent link="left_link5"/>
    <child link="left_gripper_link"/>
    <axis xyz="0.00000 1.00000 0.00000"/>
  </joint>
  <joint name="left_gripper_sub" type="prismatic">
    <limit lower="-0.01000" upper="0.01900" effort="1" velocity="4.80000"/>
    <mimic multiplier="1.00000" offset="0.00000" joint="left_gripper"/>
    <origin rpy="-0.00000 0.00000 -0.00000" xyz="0.08170 -0.02100 0.00000"/>
    <parent link="left_link5"/>
    <child link="left_gripper_link_sub"/>
    <axis xyz="0.00000 -1.00000 0.00000"/>
  </joint>
  <joint name="left_joint1" type="revolute">
    <limit lower="-2.82743" upper="2.82743" effort="1" velocity="4.80000"/>
    <origin rpy="3.14159 1.57080 -0.00000" xyz="-0.09700 -0.00000 -0.01200"/>
    <parent link="maverick"/>
    <child link="left_link2"/>
    <axis xyz="0.00000 0.00000 1.00000"/>
  </joint>
  <joint name="left_joint2" type="revolute">
    <limit lower="-1.79071" upper="1.57080" effort="1" velocity="4.80000"/>
    <origin rpy="-0.00000 0.00000 -0.00000" xyz="-0.00000 -0.00000 0.05950"/>
    <parent link="left_link2"/>
    <child link="left_link3"/>
    <axis xyz="0.00000 1.00000 0.00000"/>
  </joint>
  <joint name="left_joint3" type="revolute">
    <limit lower="-0.94248" upper="1.38230" effort="1" velocity="4.80000"/>
    <origin rpy="-0.00000 0.00000 -0.00000" xyz="0.02400 -0.00000 0.12800"/>
    <parent link="left_link3"/>
    <child link="left_link4"/>
    <axis xyz="0.00000 1.00000 0.00000"/>
  </joint>
  <joint name="left_joint4" type="revolute">
    <limit lower="-1.79071" upper="2.04204" effort="1" velocity="4.80000"/>
    <origin rpy="-0.00000 0.00000 -0.00000" xyz="0.12400 -0.00000 0.00000"/>
    <parent link="left_link4"/>
    <child link="left_link5"/>
    <axis xyz="0.00000 1.00000 0.00000"/>
  </joint>
  <joint name="maverick_to_body" type="fixed">
    <origin rpy="0.00000 0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
    <parent link="body"/>
    <child link="maverick"/>
  </joint>
  <joint name="right_end_effector_joint" type="fixed">
    <origin rpy="-0.00000 0.00000 0.00000" xyz="0.12600 0.00000 -0.00000"/>
    <parent link="right_link5"/>
    <child link="right_end_effector_link"/>
  </joint>
  <joint name="right_gripper" type="prismatic">
    <limit lower="-0.01000" upper="0.01900" effort="1" velocity="4.80000"/>
    <origin rpy="-0.00000 0.00000 0.00000" xyz="0.08170 0.02100 -0.00000"/>
    <parent link="right_link5"/>
    <child link="right_gripper_link"/>
    <axis xyz="0.00000 1.00000 0.00000"/>
  </joint>
  <joint name="right_gripper_sub" type="prismatic">
    <limit lower="-0.01000" upper="0.01900" effort="1" velocity="4.80000"/>
    <mimic multiplier="1.00000" offset="0.00000" joint="right_gripper"/>
    <origin rpy="-0.00000 0.00000 0.00000" xyz="0.08170 -0.02100 -0.00000"/>
    <parent link="right_link5"/>
    <child link="right_gripper_link_sub"/>
    <axis xyz="0.00000 -1.00000 0.00000"/>
  </joint>
  <joint name="right_joint1" type="revolute">
    <limit lower="-2.82743" upper="2.82743" effort="1" velocity="4.80000"/>
    <origin rpy="-0.00000 1.57080 -0.00000" xyz="0.09700 -0.00000 -0.01200"/>
    <parent link="maverick"/>
    <child link="right_link2"/>
    <axis xyz="0.00000 0.00000 1.00000"/>
  </joint>
  <joint name="right_joint2" type="revolute">
    <limit lower="-1.79071" upper="1.57080" effort="1" velocity="4.80000"/>
    <origin rpy="-0.00000 0.00000 0.00000" xyz="0.00000 0.00000 0.05950"/>
    <parent link="right_link2"/>
    <child link="right_link3"/>
    <axis xyz="0.00000 1.00000 0.00000"/>
  </joint>
  <joint name="right_joint3" type="revolute">
    <limit lower="-0.94248" upper="1.38230" effort="1" velocity="4.80000"/>
    <origin rpy="-0.00000 0.00000 0.00000" xyz="0.02400 0.00000 0.12800"/>
    <parent link="right_link3"/>
    <child link="right_link4"/>
    <axis xyz="0.00000 1.00000 0.00000"/>
  </joint>
  <joint name="right_joint4" type="revolute">
    <limit lower="-1.79071" upper="2.04204" effort="1" velocity="4.80000"/>
    <origin rpy="-0.00000 0.00000 0.00000" xyz="0.12400 0.00000 -0.00000"/>
    <parent link="right_link4"/>
    <child link="right_link5"/>
    <axis xyz="0.00000 1.00000 0.00000"/>
  </joint>
  <link name="body">
    <collision name="Cube">
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.00000 0.00000 -0.22261"/>
      <geometry>
        <box size="0.16000 0.06000 0.60000"/>
      </geometry>
    </collision>
  </link>
  <link name="left_end_effector_link">
    <inertial>
      <inertia ixx="0.00100" ixy="0.00000" ixz="0.00000" iyy="0.00100" iyz="0.00000" izz="0.00100"/>
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
      <mass value="0.00100"/>
    </inertial>
    <visual name="left_end_effector_link_visual">
      <origin rpy="-0.00000 0.00000 -0.00000" xyz="0.00000 -0.00000 0.00000"/>
      <material name="red"/>
      <geometry>
        <box size="0.01000 0.01000 0.01000"/>
      </geometry>
    </visual>
  </link>
  <link name="left_gripper_link">
    <collision name="left_gripper_link_collision">
      <origin rpy="-0.00000 -0.00000 -0.00000" xyz="0.00000 0.00000 -0.00000"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link_grip_l.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </collision>
    <inertial>
      <inertia ixx="0.00100" ixy="0.00000" ixz="0.00000" iyy="0.00100" iyz="0.00000" izz="0.00100"/>
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
      <mass value="0.00100"/>
    </inertial>
    <visual name="left_gripper_link_visual">
      <origin rpy="-0.00000 -0.00000 -0.00000" xyz="0.00000 0.00000 -0.00000"/>
      <material name="grey"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link_grip_l.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </visual>
  </link>
  <link name="left_gripper_link_sub">
    <collision name="left_gripper_link_sub_collision">
      <origin rpy="-0.00000 -0.00000 -0.00000" xyz="-0.00000 0.00000 -0.00000"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link_grip_r.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </collision>
    <inertial>
      <inertia ixx="0.00100" ixy="0.00000" ixz="0.00000" iyy="0.00100" iyz="0.00000" izz="0.00100"/>
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
      <mass value="0.00100"/>
    </inertial>
    <visual name="left_gripper_link_sub_visual">
      <origin rpy="-0.00000 -0.00000 -0.00000" xyz="-0.00000 0.00000 -0.00000"/>
      <material name="grey"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link_grip_r.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </visual>
  </link>
  <link name="left_link2">
    <collision name="left_link2_collision">
      <origin rpy="-0.00000 -0.00000 -0.00000" xyz="-0.00000 -0.00000 0.01900"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link2.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </collision>
    <inertial>
      <inertia ixx="0.00003" ixy="-0.00000" ixz="-0.00000" iyy="0.00003" iyz="0.00000" izz="0.00002"/>
      <origin rpy="0.00000 0.00000 0.00000" xyz="-0.00030 0.00054 0.04743"/>
      <mass value="0.09841"/>
    </inertial>
    <visual name="left_link2_visual">
      <origin rpy="-0.00000 -0.00000 -0.00000" xyz="-0.00000 -0.00000 0.01900"/>
      <material name="grey"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link2.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </visual>
  </link>
  <link name="left_link3">
    <collision name="left_link3_collision">
      <origin rpy="-0.00000 -0.00000 -0.00000" xyz="-0.00000 -0.00000 0.00000"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link3.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </collision>
    <inertial>
      <inertia ixx="0.00033" ixy="-0.00000" ixz="-0.00004" iyy="0.00034" iyz="-0.00000" izz="0.00006"/>
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.01031 0.00038 0.10170"/>
      <mass value="0.13851"/>
    </inertial>
    <visual name="left_link3_visual">
      <origin rpy="-0.00000 -0.00000 -0.00000" xyz="-0.00000 -0.00000 0.00000"/>
      <material name="grey"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link3.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </visual>
  </link>
  <link name="left_link4">
    <collision name="left_link4_collision">
      <origin rpy="-0.00000 -0.00000 -0.00000" xyz="0.00000 -0.00000 0.00000"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link4.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </collision>
    <inertial>
      <inertia ixx="0.00003" ixy="-0.00000" ixz="-0.00000" iyy="0.00024" iyz="0.00000" izz="0.00025"/>
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.09091 0.00039 0.00022"/>
      <mass value="0.13275"/>
    </inertial>
    <visual name="left_link4_visual">
      <origin rpy="-0.00000 -0.00000 -0.00000" xyz="0.00000 -0.00000 0.00000"/>
      <material name="grey"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link4.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </visual>
  </link>
  <link name="left_link5">
    <collision name="left_link5_collision">
      <origin rpy="-0.00000 -0.00000 -0.00000" xyz="-0.00000 -0.00000 0.00000"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link5.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </collision>
    <inertial>
      <inertia ixx="0.00008" ixy="0.00000" ixz="-0.00000" iyy="0.00008" iyz="0.00000" izz="0.00009"/>
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.04421 0.00000 0.00891"/>
      <mass value="0.14328"/>
    </inertial>
    <visual name="left_link5_visual">
      <origin rpy="-0.00000 -0.00000 -0.00000" xyz="-0.00000 -0.00000 0.00000"/>
      <material name="grey"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link5.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </visual>
  </link>
  <link name="maverick">
    <collision name="left_base_link_collision">
      <origin rpy="3.14159 1.57080 -0.00000" xyz="-0.08000 0.00000 0.00000"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link1.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </collision>
    <collision name="right_base_link_collision">
      <origin rpy="-0.00000 1.57080 -0.00000" xyz="0.08000 -0.00000 0.00000"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link1.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </collision>
    <inertial>
      <inertia ixx="0.00003" ixy="0.00000" ixz="-0.00000" iyy="0.00004" iyz="0.00000" izz="0.00004"/>
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.00000 -0.00000 -0.00031"/>
      <mass value="0.15824"/>
    </inertial>
    <visual name="left_base_link_visual">
      <origin rpy="3.14159 1.57080 -0.00000" xyz="-0.08000 0.00000 0.00000"/>
      <material name="grey"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link1.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </visual>
    <visual name="right_base_link_visual">
      <origin rpy="-0.00000 1.57080 -0.00000" xyz="0.08000 -0.00000 0.00000"/>
      <material name="grey"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link1.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </visual>
  </link>
  <link name="right_end_effector_link">
    <inertial>
      <inertia ixx="0.00100" ixy="0.00000" ixz="0.00000" iyy="0.00100" iyz="0.00000" izz="0.00100"/>
      <origin rpy="0.00000 0.00000 0.00000" xyz="-0.00000 0.00000 0.00000"/>
      <mass value="0.00100"/>
    </inertial>
    <visual name="right_end_effector_link_visual">
      <origin rpy="-0.00000 0.00000 0.00000" xyz="0.00000 -0.00000 0.00000"/>
      <material name="red"/>
      <geometry>
        <box size="0.01000 0.01000 0.01000"/>
      </geometry>
    </visual>
  </link>
  <link name="right_gripper_link">
    <collision name="right_gripper_link_collision">
      <origin rpy="-0.00000 0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link_grip_l.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </collision>
    <inertial>
      <inertia ixx="0.00100" ixy="0.00000" ixz="0.00000" iyy="0.00100" iyz="0.00000" izz="0.00100"/>
      <origin rpy="0.00000 0.00000 0.00000" xyz="-0.00000 0.00000 0.00000"/>
      <mass value="0.00100"/>
    </inertial>
    <visual name="right_gripper_link_visual">
      <origin rpy="-0.00000 0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
      <material name="grey"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link_grip_l.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </visual>
  </link>
  <link name="right_gripper_link_sub">
    <collision name="right_gripper_link_sub_collision">
      <origin rpy="-0.00000 0.00000 0.00000" xyz="0.00000 -0.00000 0.00000"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link_grip_r.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </collision>
    <inertial>
      <inertia ixx="0.00100" ixy="0.00000" ixz="0.00000" iyy="0.00100" iyz="0.00000" izz="0.00100"/>
      <origin rpy="0.00000 0.00000 0.00000" xyz="-0.00000 0.00000 0.00000"/>
      <mass value="0.00100"/>
    </inertial>
    <visual name="right_gripper_link_sub_visual">
      <origin rpy="-0.00000 0.00000 0.00000" xyz="0.00000 -0.00000 0.00000"/>
      <material name="grey"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link_grip_r.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </visual>
  </link>
  <link name="right_link2">
    <collision name="right_link2_collision">
      <origin rpy="-0.00000 0.00000 0.00000" xyz="0.00000 -0.00000 0.01900"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link2.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </collision>
    <inertial>
      <inertia ixx="0.00003" ixy="-0.00000" ixz="-0.00000" iyy="0.00003" iyz="0.00000" izz="0.00002"/>
      <origin rpy="0.00000 0.00000 0.00000" xyz="-0.00030 0.00054 0.04743"/>
      <mass value="0.09841"/>
    </inertial>
    <visual name="right_link2_visual">
      <origin rpy="-0.00000 0.00000 0.00000" xyz="0.00000 -0.00000 0.01900"/>
      <material name="grey"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link2.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </visual>
  </link>
  <link name="right_link3">
    <collision name="right_link3_collision">
      <origin rpy="-0.00000 0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link3.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </collision>
    <inertial>
      <inertia ixx="0.00033" ixy="-0.00000" ixz="-0.00004" iyy="0.00034" iyz="-0.00000" izz="0.00006"/>
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.01031 0.00038 0.10170"/>
      <mass value="0.13851"/>
    </inertial>
    <visual name="right_link3_visual">
      <origin rpy="-0.00000 0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
      <material name="grey"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link3.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </visual>
  </link>
  <link name="right_link4">
    <collision name="right_link4_collision">
      <origin rpy="-0.00000 0.00000 0.00000" xyz="-0.00000 0.00000 0.00000"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link4.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </collision>
    <inertial>
      <inertia ixx="0.00003" ixy="-0.00000" ixz="-0.00000" iyy="0.00024" iyz="0.00000" izz="0.00025"/>
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.09091 0.00039 0.00022"/>
      <mass value="0.13275"/>
    </inertial>
    <visual name="right_link4_visual">
      <origin rpy="-0.00000 0.00000 0.00000" xyz="-0.00000 0.00000 0.00000"/>
      <material name="grey"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link4.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </visual>
  </link>
  <link name="right_link5">
    <collision name="right_link5_collision">
      <origin rpy="-0.00000 0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link5.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </collision>
    <inertial>
      <inertia ixx="0.00008" ixy="0.00000" ixz="-0.00000" iyy="0.00008" iyz="0.00000" izz="0.00009"/>
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.04421 0.00000 0.00891"/>
      <mass value="0.14328"/>
    </inertial>
    <visual name="right_link5_visual">
      <origin rpy="-0.00000 0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
      <material name="grey"/>
      <geometry>
        <mesh filename="package://maverick/meshes/stl/chain_link5.stl" scale="0.00100 0.00100 0.00100"/>
      </geometry>
    </visual>
  </link>
  <material name="grey">
    <color rgba="0.49804 0.49804 0.49804 1.00000"/>
  </material>
  <material name="red">
    <color rgba="0.80000 0.00000 0.00000 1.00000"/>
  </material>
</robot>
